### PR TITLE
buildGoModule/buildGoPackage: Remove go references from lib,libexec

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
`buildGoModule`?

**Comment by zowoq** [comment]
If we're doing this we may as well do `sbin` and `lib`.

**Comment by zowoq** [comment]
Also:
https://github.com/NixOS/nixpkgs/blob/6eae50cca8014a2eb773d73bbe10552c60841217/pkgs/applications/virtualization/singularity/default.nix#L71

**Comment by Mic92** [comment]
> If we're doing this we may as well do `sbin` and `lib`.

I checked `sbin` but it is already moved in fixupPhase to `bin`, so I left it.

**Comment by Mic92** [comment]
> Also:
> 
> https://github.com/NixOS/nixpkgs/blob/6eae50cca8014a2eb773d73bbe10552c60841217/pkgs/applications/virtualization/singularity/default.nix#L71

done

**Comment by Mic92** [comment]
> `buildGoModule`? Nothing uses libexec.

Eventually docker and co could be migrated to it at which point it will be used.

File: pkgs/development/go-packages/generic/default.nix [hunk 1]
```diff
@@ -211,8 +211,7 @@ let
     '';
 
     preFixup = preFixup + ''
-      find $out/bin -type f -exec ${removeExpr removeReferences} '{}' + || true
-      find $out/libexec -type f -exec ${removeExpr removeReferences} '{}' + || true
+      find $out/{bin,libexec,lib} -type f 2>/dev/null | xargs -r -n 1 ${removeExpr removeReferences} || true
```
Comment (lheckemann) [inline pkgs/development/go-packages/generic/default.nix:214]: Why `-n 1`? Depending on the amount of binaries, using multiple args could save a significant amount of processes. The original version with `-exec … {} +` used multiple args too.

**Comment by Mic92** [comment]
There is also the `deepin` desktop (~ 3 packages) that could make use of it but it is marked as broken atm.

File: pkgs/development/go-packages/generic/default.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/development/go-packages/generic/default.nix:214]: I was not aware of that. Will fix.

Comment (Mic92) [reply pkgs/development/go-packages/generic/default.nix]: done.

**Comment by zowoq** [comment]
> There is also the deepin desktop (~ 3 packages) that could make use of it but it is marked as broken atm.

I'd include them anyway.

**Comment by Mic92** [comment]
done.

**Comment by Mic92** [comment]
All good?

**Review by zowoq — APPROVED** [review]

**Review by lheckemann — APPROVED** [review]

**Review by kalbasit — APPROVED** [review]

**Pull request merged**